src/error.ts

```diff
@@ -1,10 +1,33 @@
-import { bold, red } from "../deps.ts";
+// Copyright 2024 Deno Land Inc. All rights reserved. MIT license.
 
-export function printError(message: string) {
+import { bold, red } from "@std/fmt/colors";
+
+export function error(err: unknown): never {
+  const message = stringify(err);
   console.error(red(`${bold("error")}: ${message}`));
+  Deno.exit(1);
 }
 
-export function error(message: string): never {
-  printError(message);
-  Deno.exit(1);
+export function stringify(err: unknown): string {
+  if (err instanceof Error) {
+    return stringifyError(err);
+  }
+
+  if (typeof err === "string") {
+    return err;
+  }
+
+  return JSON.stringify(err, null, 2);
+}
+
+function stringifyError(err: Error): string {
+  const cause = err.cause === undefined
+    ? ""
+    : `Caused by ${stringify(err.cause)}`;
```

Comment:
Maybe add the new line in here, this way we don't get an empty line if there's no cause.

Having said that, being a CLI tool, deployctl is sensible to changes in the formatting of the error messages. I'm a bit concerned that adding this newline will break the formatting here and there inadvertently. Maybe we could use  parenthesis `(${cause})` instead of new line `\n${cause}`?

Reply:
Can you elaborate a bit more on why CLI tools are sensible to the change of error message formats? 

> Maybe we could use parenthesis (${cause}) instead of new line \n${cause}?

Parenthesis sounds good, but the next problem is that `cause` itself might contain newline characters in it. I'll try to replace them (if any) with white spaces.

Reply:
There are places where the error is not fatal, and should be shown to the user intertwined with regular output. If the error has multiple lines it breaks the output layout. One example is https://github.com/denoland/deployctl/pull/332#discussion_r1806424870

Reply:
That makes lots of sense. I'll create another function to stringify errors in a simple, one-line format rather than full verbose ones so we can use it in non-fatal places.

Reply:
Now `stringify` function gets `verbose` option which defaults to `false`. If an error is printed when the deployctl program keeps running, simple message is displayed. Otherwise i.e. when the program exits right after the error message is printed, verbose message is generated.